Update Checkout appearance section to display WooPay checkout preview UI

lovo-h opened the pull request:
Fixes #6995 

#### Changes proposed in this Pull Request

<!--
Title: A descriptive yet concise title.
-->

<!--
Description: Write a summary of this PR. As you write your summary, please look at each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Could you let me know why you chose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

This PR updates **WooCommerce > Settings > Payments > WooPayments > WooPay > Checkout appearance** so that the WooPay checkout preview resembles the WooPay checkout page.

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Test: Store name and logo
* As a merchant, navigate to **WooCommerce > Settings > Payments > WooPayments > WooPay > Checkout appearance**.
* Click the `Upload custom logo` button and upload one of the accepted image file types (e.g. HEIC, WEBP, SVG, PNG, or JPG).
* Ensure the preview now has the custom logo displayed.
* Delete the logo by clicking the trash icon.
* Ensure the preview now has the store's name displayed.

https://github.com/Automattic/woocommerce-payments/assets/6844516/6ba8a247-ba69-47a5-a712-985e8181d6fc


Test: Checkout privacy policy
* As a merchant, navigate to **WooCommerce > Settings > Payments > WooPayments > WooPay > Checkout appearance**.
* Add some text to the `Policies and custom text` field.
* Ensure the text changes are displayed in the preview. Also, ensure spacing is adjusted accordingly.


https://github.com/Automattic/woocommerce-payments/assets/6844516/396f180f-a2a9-4bce-9bcd-4372dff12a96


---

**Screen Caps**

Mobile
![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/5c60dfe5-0633-4158-963e-c795aa95b5db)

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/1a508946-33c9-4d61-80dd-562019a2c7da)

Desktop

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/8d488777-486e-403d-93e3-81bff37a3e29)

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/b18e760e-054b-4d44-b7ca-00923b0eaa83)


-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

lovo-h committed:
Reword 'Store logo on checkout' to 'Checkout logo'

lovo-h committed:
Reword Store logo's help-text: 'Upload a custom logo. For best...'

lovo-h committed:
Add horizontal line between preview section and preview configurations

lovo-h committed:
Remove fade effect between preview and configuration section

lovo-h committed:
Update SVG for WooPay checkout preview UI

lovo-h committed:
Make preview dynamic: customMessage, storeLogo, and storeName

lovo-h committed:
Add changelog

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7511` or branch name `update/6995-woopay-checkout-appearance-preview-ui` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/6995-woopay-checkout-appearance-preview-ui) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5fdd6bceb153aa662460802232d53b9219be4d25
- Build time: 2023-10-27 20:00:30 UTC

_Note: the build is updated when a new commit is pushed to this PR._

lovo-h committed:
Ensure background is white

lovo-h committed:
Fix typo: Remove typo-ed comma

It seems like this comma was accidentally merged. This commit removes
it since it appears to be unintentional.

pierorocca commented:
@lovo-h thanks for your work on this enhancement. Super excited about this. It'll be a good improvement for merchants evaluating or setting up WooPay. @nikkivias would appreciate your eyes on this before it's pushed to production.

Desktop: Looks good. It's a bit hard to tell from the video, is the preview policy text legible in your opinion?  Thoughts on the button loading component? Is that a standard loading component @nikkivias ?

Regarding the logo, the image formats I believe I provided the copy for that. Do we in fact support all of those image formats? Double checking.

Mobile: Is it your phone font setting that makes the input text appear to be oversized? The preview also looks super tiny. Is it possible to increase the width of the preview to be same width at the section below it?

pierorocca commented:
@nikkivias would you recommend that the preview on desktop also make use of the extra available width?

cesarcosta99 commented on client/settings/express-checkout-settings/woopay-settings.js:
```diff
@@ -182,24 +182,27 @@ const WooPaySettings = ( { section } ) => {
+						style={ {
+							margin: '0 -1.5rem 1.0rem -1.5rem',
+							width: 'calc(100% + 3rem)',
+						} }
```
Should we make use of classes to maintain consistency in styling?

lovo-h replied:
Good catch. Fixed in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/96867703a8608c5b941a1bd5f22002e4f469168d.

cesarcosta99 commented on client/settings/express-checkout-settings/woopay-preview.js:
```diff
@@ -10,215 +10,264 @@ import { decodeEntities } from '@wordpress/html-entities';
+const svgParams = Object.freeze( {
+	CUSTOM_MSG_HEIGHT: 434,
+	DEFAULT_HEIGHT: 396,
+} );
```
We could very likely reduce build size by using constants with plain values here. The value of using `Object.freeze` is to have the object immutable, but we can achieve the same by having a plain `const`.

```js
const CUSTOM_MSG_HEIGHT = 434;
const DEFAULT_HEIGHT = 396;
```

lovo-h replied:
Good call. Fixed in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/6799d61f3e88c371e532fb0b4786139935aa41c3.

cesarcosta99 commented on client/settings/express-checkout-settings/woopay-settings.js:
```diff
@@ -182,24 +182,27 @@ const WooPaySettings = ( { section } ) => {
 							accept="image/png, image/jpeg"
 							disabled={ false }
 							help={ __(
-								'Use a custom logo to WooPay if the one taken from your store doesn’t look right.' +
-									' For best results, upload a high-resolution horizontal image' +
-									' with white or transparent background.',
+								'Upload a custom logo. For best results, upload a horizontal' +
+									' image with white or transparent background. HEIC, WEBP, SVG, PNG,' +
+									' or JPG. Recommended width: 512 pixels minimum.',
```
When I click to upload a logo, I can't select `.heic`, `.webp` or `.svg` formats. So we either need to include them in the `accept` property, or update the copy saying we only accept `.png` or `.jpg` images.

lovo-h replied:
Good catch. It looks like we plan on moving forward with only supporting `JPG` and `PNG` for now (https://github.com/Automattic/woocommerce-payments/pull/7511#issuecomment-1771899300). Updated the text copy in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/28f508e049c3497b71ed765b5d04647395208405:
```
Upload a custom logo. For best results, upload a horizontal 
image with white or transparent background. PNG or JPG. 
Recommended width: 512 pixels minimum.
```

cesarcosta99 reviewed (changes requested):
Changes look good and tests well 😎  I left a few nitpick comments but the only blocker here is the image formats we accept for the logo.

- The preview of the WooPay checkout looks accurate ✅ 
- Both the logo and custom text are correctly applied at the WooPay checkout ✅ 
- Removing the logo reverts to the store name at the WooPay checkout ✅ 

Another suggestion, if that's not a big deal, would be to stretch the preview to the content area in the mobile view, as follows:

<img width="376" alt="Screenshot 2023-10-19 at 19 00 44" src="https://github.com/Automattic/woocommerce-payments/assets/10233985/2093beba-2921-44b3-833e-2302ab52e261">

lovo-h commented:
> Desktop: Looks good. It's a bit hard to tell from the video, is the preview policy text legible in your opinion?

@pierorocca, here's a better view of the desktop version:

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/82f7363e-fc90-442a-b203-80b01d0f5954)

The text is barely legible but does indicate the general shape of the text and how it would appear on the checkout page.

We may be able to increase the text size to make it more legible by reducing the horizontal white space. But in order to do this, I will need an updated Figma design; since I am using the SVG as my source. Here is what the Figma design currently looks like:

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/f2b6d688-bb18-408e-8c1c-d238689e3f9d)

Source: nngbiDUWyJa7fYcpiE9Jdu-fi-5649_4058.

> Regarding the logo, the image formats I believe I provided the copy for that. Do we in fact support all of those image formats? Double checking.

Good question. Currently, we only support `PNG` and `JPG`. Did we want to rephrase the text, or do we want to place effort in supporting the other image formats?

> Mobile: Is it your phone font setting that makes the input text appear to be oversized?

Actually, in this case, I was using Chrome's Dev Tools to preview how it would look on a mobile device. I agree though, the input text does look oversized. Here's how it appears on my mobile device (iPhone):

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/58001a29-fc3c-4515-b227-fda4e19f50a1)

It looks about the same 🤔 

> The preview also looks super tiny. Is it possible to increase the width of the preview to be same width at the section below it?

I agree, that would be a good solution. At the moment, I am importing the design layout as an SVG, from the Figma designs. So, in order to update the preview on mobile, I will need new Figma designs (tailored for mobile). 

Also, for the Desktop version, if we are able to update the Figma designs so that we use more of the empty space, we may be able to make the text more legible. 

Once the designs have been updated, I can import and use the new SVGs. @nikkivias, are we able to make these changes on Figma?

pierorocca commented:
Sorry for the brevity as I'm on mobile on meet-up. Let's adjust the copy to match existing capability and we can decide to support additional formats as a new enhancement issue.

pierorocca commented:
> The text is barely legible but does indicate the general shape of the text and how it would appear on the checkout page.

A key objective of the preview is to allow the merchant to view the wording of the policy text. I've left a comment on the Figma file outlining the problem with legibility, especially on mobile.

pierorocca commented:
The mobile input font size, what do you believe is causing it to be so huge relative to the size on desktop?

lovo-h committed:
Rephrase WooPayFileUpload help text: indicate support for JPG, PNG only

lovo-h committed:
WooPaySettings: Move inline CSS into a CSS file

lovo-h committed:
Refactor: Replace `Object.freeze()` with constants

This will help reduce the file build size.

lovo-h committed:
Fix: Text area font size to be 0.8125rem for both desktop & mobile

lovo-h commented:
> Let's adjust the copy to match existing capability and we can decide to support additional formats as a new enhancement issue.

@pierorocca, the copy has been updated in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/28f508e049c3497b71ed765b5d04647395208405:
```
Upload a custom logo. For best results, upload a horizontal 
image with white or transparent background. PNG or JPG. 
Recommended width: 512 pixels minimum.
```

Should we rephrase things to the following:
```
Upload a custom logo. Upload a horizontal image with a white
or transparent background for best results. Use a PNG or JPG
image format. Recommended width: 512 pixels minimum.
```


> A key objective of the preview is to allow the merchant to view the wording of the policy text. I've left a comment on the Figma file outlining the problem with legibility, especially on mobile.

Awesome! Thank you 🙏 

> The mobile input font size, what do you believe is causing it to be so huge relative to the size on desktop?

Mobile (before)

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/2e5b6d68-3e68-48e0-af06-6938e44d8c4e)

This was an existing issue in the mobile version. The mobile version's input font size was set to `16 px`. This has been fixed to match the desktop designs in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/306edcd1c3421998aa39ecb65d23598aa4f4e960:

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/f16f75a6-4c48-42bd-8714-2f76e67152a0)

Desktop (for comparison)

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/570c9a76-12f2-464c-94a5-cb29910710a1)

In the desktop version, the input font size was already set to `13 px` (as indicated in the designs).

pierorocca commented:
Yes your phrasing is better. Nice suggestion.

cesarcosta99 reviewed (approved):
Thanks for addressing the changes. From now on we'll be just addressing some design aspects of it, so I'm approving it.

pierorocca commented:
Thanks @lovo-h the font changes look good. Thanks for making that change.

So we're waiting on the design team to provide a larger sized preview SVG to which we'll increase the preview font size. This is slated for 6.7.0 so we have a few days still?

lovo-h committed:
Rephrase WooPayFileUpload help text

lovo-h commented:
> Yes your phrasing is better. Nice suggestion.

Thank you. Updated in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/c4185128123361fc74201b4b836a15f8ffcd2d85.

> So we're waiting on the design team to provide a larger sized preview SVG to which we'll increase the preview font size. This is slated for 6.7.0 so we have a few days still?

@pierorocca, that's correct, on both counts. 

Previously, we were dropping in the design as an SVG element. However, we may be able to skip the SVG element and try to recreate it in code. This will allow us to make the WooPay checkout preview more responsive.

pierorocca commented:
> Previously, we were dropping in the design as an SVG element. However, we may be able to skip the SVG element and try to recreate it in code. This will allow us to make the WooPay checkout preview more responsive.

I actually quite like that because it'll always be in-synch and reflect reality. Not too far from now, we'll be looking at what Global theme settings we can pull in so that there's a bit more brand consistency between the store and WooPay. Your approach would be more compatible and aligned.

lovo-h commented:
> I actually quite like that because it'll always be in-synch and reflect reality. Not too far from now, we'll be looking at what Global theme settings we can pull in so that there's a bit more brand consistency between the store and WooPay. Your approach would be more compatible and aligned.

Great to hear! I'll go ahead and remove the blocked status and refactor things accordingly 👍

lovo-h committed:
Add SVG icon: `lock-keyhole`

lovo-h committed:
Refactor WooPay checkout preview UI: from SVG to HTMl

lovo-h commented:
> Thanks for addressing the changes. From now on we'll be just addressing some design aspects of it, so I'm approving it.

@cesarcosta99, thank you for the code review. But since we have refactored things, could you please review the changes, once more?

lovo-h commented:
@pierorocca, here is what the WooPay checkout preview UI looks like now:

https://github.com/Automattic/woocommerce-payments/assets/6844516/a51b7fb7-a670-49ab-99d0-f53f8aa9f416

Things to note:
* The "Secure and encrypted with AES-256" text and the store name both get truncated with an ellipsis in windows with narrow widths. This is intentional to prevent the layout from becoming skewed.
* The **Policies and custom text** gets truncated and hidden when the text is too long and/or the window width is too narrow. This is intentional to prevent the layout from becoming skewed.
* The **Policies and custom text** is more legible, but we can adjust the font size slightly if it needs improvement.

---

Here is what the behavior looks like when we update the **Policies and custom text** and the **Checkout logo**:

https://github.com/Automattic/woocommerce-payments/assets/6844516/8f58c7c9-6b95-438b-9ff3-2b9dadf1b8cc

cc @nikkivias

pierorocca commented:
I ran upstairs to see this on a larger screen and because I was so excited to see it. Looks pretty awesome Hector. The only thing I can think of @nikkivias is does a merchant know and understand that it's a preview? Does it need different or additional titling? Thoughts on swapping the setting and preview section order so the preview is after the settings? That's consistent with the Button previews in which the settings are presented first and then the button preview is shown below the last setting.

pierorocca commented:
To confirm, if the design language and styling is updated in the near future, this preview will inherit those changes without dev?

lovo-h commented:
> To confirm, if the design language and styling is updated in the near future, this preview will inherit those changes without dev?

Not at the moment, but the current refactor allows us to easily hook into the global theme settings (once they're available), which will ultimately allow us to update the design language and styling without needing a dev (this assumes that the global theme settings will be configurable through the UI).

Previously, since we were using an SVG, we would have required the design team and a dev to make the changes.

lovo-h committed:
Add "Internal dependencies." comment back in

pierorocca commented:
Beyond global themes settings, if we on the WooPay team made a component design change to checkout p1698084577493739-slack-C043JN7AJHY would the preview inherit the changes?

lovo-h commented:
> Beyond global themes settings, if we on the WooPay team made a component design change to checkout p1698084577493739-slack-C043JN7AJHY would the preview inherit the changes?

Ah, I see. No, the preview would not inherit those changes since it's the page layout that's being updated. 

The good news is that it appears like it's mainly the left column that's undergoing significant changes. So, we would be able to reuse most of the changes made here.

nikivias commented:
Hey team. The update on the preview looks awesome! I love how smoothly it responds @lovo-h 

> does a merchant know and understand that it's a preview? Does it need different or additional titling? Thoughts on swapping the setting and preview section order so the preview is after the settings? 

@pierorocca The page is quite abstracted (grey boxes) which is good in conveying that it's a preview, however it is the norm to add a small title that says 'Preview'. 'Checkout appearance' as a title makes sense too.

You make a good point about swapping the order between the setting and the preview as this is more the conventional way, and more important that pattern already exists in this actual page. (Although I can't seem to find the button settings you are referring to) I did a bit of digging in design files for Blaze, and other ads flows in Woo— we also show the settings first, then preview with a preview title.

Below I've mocked up what this might look like. [See it on Figma](https://www.figma.com/file/nngbiDUWyJa7fYcpiE9Jdu/WooPay-Design?type=design&node-id=8719%3A72567&mode=design&t=gC7kJTxowDi4SMRK-1)

<img width="1440" alt="Checkout privacy policy - Settings saved" src="https://github.com/Automattic/woocommerce-payments/assets/99638528/18148464-0b53-41f0-ad9f-3a4f97de7e13">

pierorocca commented:
That looks terrific @nikkivias. 

Here's the other preview I was referring to. The Express Payments checkout button has settings associated with it and we display a preview of the button appearance below. Note because I'm on a work profile on Chrome, GPay doesn't render but it would if I was signed in on my personal Chrome account.

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/b61a8e6e-6476-45ab-88f1-470aca0bb5da)

nikivias commented:
That button setting is looking a little too tall and could use some reworking too

bborman22 commented:
I'm loving the direction this is taking! One thing to keep in mind as we review this PR is that we'll need it merged tomorrow to make it into 6.7. @lovo-h I don't have a solid grasp on the scope of any given request on this issue, but we'll want to keep that deadline in mind and may need to make some trade offs to be sure we hit that.

pierorocca commented:
@nikkivias there's an open enhancement issue resulting from a recent user flow review. Welcome your feedback on the proposed design. https://github.com/Automattic/woocommerce-payments/issues/6470

lovo-h committed:
Remove divider

lovo-h committed:
Reorder WooPay Checkout preview with WooPay Checkout settings

lovo-h commented:
@pierorocca, @nikkivias, here is how the latest changes appear:

**When resizing the window**

https://github.com/Automattic/woocommerce-payments/assets/6844516/f41f6c9f-ee4d-4092-902a-016221af4ccf


**When updating the Policies and custom text & the Checkout logo**

https://github.com/Automattic/woocommerce-payments/assets/6844516/f2edc17c-5b4d-4638-ac5e-9fbf993c5062

pierorocca commented:
Looks fantastic. Thanks all for the iterative collaboration. It's amazing to see how small adjustments have such huge impact on the final result. LGTM.

nikivias commented:
That was amazingly quick! Looking good! I would just note to add that bit of shadow and border radius I added to the actual preview.

lovo-h committed:
Add box-shadow to .preview-layout

lovo-h committed:
Add border-radius to .preview-layout

lovo-h commented:
> That was amazingly quick! Looking good! I would just note to add that bit of shadow and border radius I added to the actual preview.

Great catch @nikkivias. The shadow and the border-radius have been added:

![image](https://github.com/Automattic/woocommerce-payments/assets/6844516/6282656d-293c-47f1-a506-cf99e8df259a)

cesarcosta99 reviewed (changes requested):
- Everything is working as in the first review ✅
- The checkout preview is responsive ✅

I've found two issues that we might want to consider addressing still in this PR.

1. In any viewport, the height of the preview is bigger than the container when the custom text is empty, making the box-shadow overflow the section.

![Oct-27-2023 13-21-59](https://github.com/Automattic/woocommerce-payments/assets/10233985/56a4ebf7-1ab2-4f1a-958d-a81ecec91e6d)

2. Should we account for line breaks here? The line breaks present in the textarea are not reflected at the WooPay checkout or the preview. This is slightly unrelated to the goal of this PR, so I think it'd be ok to address in a different PR. cc @pierorocca 

<img width="717" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/10233985/00697110-a28a-4a3c-a1b9-618923081f64">

pierorocca commented:
> In any viewport, the height of the preview is bigger than the container when the custom text is empty, making the box-shadow overflow the section.

Nice catch! All of us missed that in the demos. That definitely should be resolved prior to merge.

> Should we account for line breaks here? The line breaks present in the textarea are not reflected at the WooPay checkout or the preview. This is slightly unrelated to the goal of this PR, so I think it'd be ok to address in a different PR. cc 

This can wait until we either support markdown or add rich text capable editor. The key thing is that we can surface the store policies. Longer term I'd like to see 1:1 alignment with the checkout policy block + some minimal custom rich text via an easy to use wysiwyg editor.

lovo-h committed:
Fix box overflow: replace height values with 'rem' as opposed to '%'

lovo-h committed:
Allow custom message element to be more responsive

lovo-h commented:
> 1. In any viewport, the height of the preview is bigger than the container when the custom text is empty, making the box-shadow overflow the section.

@cesarcosta99, great catch! Fixed in https://github.com/Automattic/woocommerce-payments/pull/7511/commits/4692f5bdad574612943d4ba221c415022b679a1b.

lovo-h commented:
Here's a demo with the fix applied:

https://github.com/Automattic/woocommerce-payments/assets/6844516/b367a028-42fc-4ce5-bea6-2dddf0f5af9c

cesarcosta99 reviewed (approved):
The shadow of the checkout preview is looking good now 🚀

lovo-h committed:
Merge branch 'develop' into update/6995-woopay-checkout-appearance-preview-ui

pierorocca commented:
@lovo-h it appears that the shortcode links don't render in the preview. Instead [privacy_policy_link] and [terms_of_service_link] render. We missed that as it wasn't a part of any of the previews. Thanks @csmcneill for catching this pre-release.

Can you replicate this on your site?

pierorocca commented:
Also this is out of scope of this issue, Chris also makes a good point that on core the shortcode links are [privacy_policy] and [terms]. Would it make sense to keep consistent?

lovo-h commented:
> @lovo-h it appears that the shortcode links don't render in the preview. Instead [privacy_policy_link] and [terms_of_service_link] render. We missed that as it wasn't a part of any of the previews. Thanks @csmcneill for catching this pre-release.
> 
> Can you replicate this on your site?

@pierorocca, I was able to replicate this on my site. A fix has been created in #7601.

> Also this is out of scope of this issue, Chris also makes a good point that on core the shortcode links are [privacy_policy] and [terms]. Would it make sense to keep consistent?

I went ahead and made these shortcodes consistent in the **Policies and custom text** input box (for the WooPay checkout preview). But we may still need to create a separate issue for making them consistent in the **Appearance > Customize > WooCommerce > Checkout > Privacy policy / Terms and conditions** input boxes.

pierorocca commented:
Thanks @lovo-h for the quick fix. Will that make 6.7.0 or are we past that window?

> I went ahead and made these shortcodes consistent in the Policies and custom text input box (for the WooPay checkout preview). But we may still need to create a separate issue for making them consistent in the Appearance > Customize > WooCommerce > Checkout > Privacy policy / Terms and conditions input boxes.

Based on the demo in #7601 am I understanding correctly that for now we'll still use the _link version and we'll open a new issue to align between core and WooPay?

lovo-h commented:
Piero, we discussed this over Slack but to reiterate here:

> Thanks @lovo-h for the quick fix. Will that make 6.7.0 or are we past that window?

It looks like WooPayments 6.7.0 has already been released. So, this will have to be included in the next release: WooPayments 6.8.0 (Nov 22).

> Based on the demo in #7601 am I understanding correctly that for now we'll still use the _link version and we'll open a new issue to align between core and WooPay?

Initially, that was the plan but since the changes were small enough, I went ahead and added it to #7601 in 
[827f47e](https://github.com/Automattic/woocommerce-payments/pull/7601/commits/827f47e05dd12ab2428098f1f0f5821b32326e14).
